test/unit/org/apache/cassandra/cql3/ast/CollectionAccess.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.ast;
+
+import java.util.stream.Stream;
+
+import org.apache.cassandra.db.marshal.AbstractType;
+
+public class CollectionAccess implements ReferenceExpression
+{
+    private final ReferenceExpression column;
+    private final ReferenceExpression element;
+    private final AbstractType<?> type;
+
+    public CollectionAccess(ReferenceExpression column, ReferenceExpression element, AbstractType<?> type)
+    {
+        this.column = column;
+        this.element = element;
+        this.type = type;
+    }
+
+    @Override
+    public void toCQL(StringBuilder sb, int indent)
+    {
+        column.toCQL(sb, indent);
+        sb.append('[');
```

Comment:
I feel like this isn't generic... when the AST was first made some of the collection stuff was not done yet, so may want to relook into this